### PR TITLE
first submission

### PR DESCRIPTION
**Pull request by kokox002**

### CODE REVIEW

File: app/concepts/todo_list/find.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class TodoList::Find < Trailblazer::Operation
+
+  def process(params)
+      model.user = params[:current_user] == nil ? User::Find.(params).model : params[:current_user]
+      model.save!
+  end
+
+  def model!(params)
+    if params[:todo_list_id].blank?
```
Comment (kennethcalamay) [inline app/concepts/todo_list/find.rb:8]: What happens when params[:todo_list_id] is not blank?

File: app/concepts/user/find.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class User::Find < Trailblazer::Operation
```
Comment (kennethcalamay) [inline app/concepts/user/find.rb:1]: Nice!

File: app/concepts/user/find.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+  def process(params)
+    model.save!
```
Comment (kennethcalamay) [inline app/concepts/user/find.rb]: May I know why we're saving the model here?

File: spec/concepts/user/find_spec.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe TodoList::Find do
```
Comment (kennethcalamay) [inline spec/concepts/user/find_spec.rb]: We're inside `user/find_spec`. May I know why we're specifying `TodoList::Find`?

File: spec/concepts/user/find_spec.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe User::Find do
+  it "finds a user guest" do
+    todo = Todo::Create.(todo: {title: "Testing helps!"}).model
+    expect(todo.list.user.fullname).to eq("Guest")
+  end
```
Comment (kennethcalamay) [inline spec/concepts/user/find_spec.rb:7]: Can you check if you have forgotten to implement User::Find spec? We're currently testing Todo::Create inside User::Find spec.

File: spec/concepts/todo_list/find_spec.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe TodoList::Find do
+  it "finds a todo default list" do
+    todo = Todo::Create.(todo: {title: "Testing helps!"}).model
+    expect(todo.list.name).to eq("Default To-do List")
```
Comment (kennethcalamay) [inline spec/concepts/todo_list/find_spec.rb]: This too, we need the correct specs for TodoList::Find

File: app/concepts/todo_list/find.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+      model.user = params[:current_user] == nil ? User::Find.(params).model : params[:current_user]
+  end
+
+  def model!(params)
+    if params[:todo_list_id].blank?
+      TodoList.find_or_create_by(name: "Default To-do List")
+    else
+      TodoList.find(id: params[:todo_list_id])
+    end
+  end
```
Comment (kennethcalamay) [inline app/concepts/todo_list/find.rb:13]: `model!` method looks good!

File: app/concepts/todo_list/find.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+  def process(params)
+      model.user = params[:current_user] == nil ? User::Find.(params).model : params[:current_user]
+  end
+
+  def model!(params)
+    if params[:todo_list_id].blank?
+      TodoList.find_or_create_by(name: "Default To-do List")
+    else
+      TodoList.find(params[:todo_list_id])
```
Comment (kennethcalamay) [inline app/concepts/todo_list/find.rb:11]: Should we scope this from current_user? i.e. `current_user.todo_lists.find`

File: app/concepts/todo/create.rb [hunk 9]
```diff
@@ -1,50 +1,17 @@
 class Todo::Create < Trailblazer::Operation
 
-  # Logic for creating a TODO
```
Comment (xirukitepe) [inline app/concepts/todo/create.rb:3]: just wanna share this good article about comments https://robots.thoughtbot.com/letting-your-code-speak-for-itself . Good work btw. :+1:

File: app/concepts/todo/create.rb [hunk 10]
```diff
@@ -1,50 +1,17 @@
...
+  end
 
-      # assign the todo_list to the user
-      todo_list.user = current_user
-      todo_list.save!
-    else # i.e. if params[:todo_list_id] is not nil
-      todo_list = TodoList.find(params[:todo_list_id])
+  def process(params)
+    validate(params[:todo]) do
+      model.list = TodoList::Find.(params).model
```
Comment (xirukitepe) [inline app/concepts/todo/create.rb:12]: you can actually use `setup_model!` in setting up things related to the `model` itself. but this is a nice interpretation of what can be done in `op`

File: app/concepts/todo/form_cell.rb [hunk 11]
```diff
@@ -21,7 +21,7 @@ def default_prompt
   end
 
   def todo_list
-    TodoList.where(params[:todo_list_id]).first
+    TodoList.where(id: params[:todo_list_id]).first
```
Comment (xirukitepe) [inline app/concepts/todo/form_cell.rb:24]: how about `find_by(id: params[:todo_list_id])` ?

File: app/concepts/todo_list/find.rb [hunk 12]
```diff
@@ -0,0 +1,15 @@
+class TodoList::Find < Trailblazer::Operation
+
+  def process(params)
+      model.user = params[:current_user] == nil ? User::Find.(params).model : params[:current_user]
```
Comment (xirukitepe) [inline app/concepts/todo_list/find.rb:4]: same comment from here https://github.com/serviceseeking/serviceseeking-test/pull/9/files#r59200987

File: spec/concepts/todo/create_spec.rb [hunk 13]
```diff
@@ -1,7 +1,8 @@
 require 'rails_helper'
 
 RSpec.describe Todo::Create do
-  it "can be spec'd" do
-    raise "Implement me"
+  it "can create a todo" do
+    todo = Todo::Create.(todo: {title: "Testing helps!"}).model
+    expect(todo.title).to eq("Testing helps!")
```
Comment (xirukitepe) [inline spec/concepts/todo/create_spec.rb:6]: would be good also if you could test the invalid / failing ones / falsy ones... but this looks good and direct :+1:

**Pull request closed without merging**